### ISSUE TITLE
No inference with best.pt

### ISSUE THREAD

**Issue by saroshpatel** [ultralytics/ultralytics#13115]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

[](url)Hello,

Here is the background to my question.

1. I am using YoloV8.2 n-model, to train 800x800 grayscale uint16 PNG images with a batch size of 64. 
2. I setup of the training for 1500 epochs. Saving every 10 epochs
3. After about 1000 epochs the training started returning NaNs for the training losses in every epoch. See Below
![Screenshot 2024-05-22 103513](https://github.com/ultralytics/ultralytics/assets/48224304/235b5bff-111b-4272-a99d-223b7f819ee3)
4. I stopped the training at around 1100 epoch.
5. I ran the validation with best.pt, and got a picture perfect confusion matrix. See below. I see the detection in the mosiac generated by the validation process
![confusion_matrix](https://github.com/ultralytics/ultralytics/assets/48224304/c175796c-c204-4a62-be5c-8e2e9f1f593e)

Q. when I use the same best.pt to run inference on the same validation set images I get NO DETECTIONS for all the images. Simply detections.

how is model.val( ) getting detections as seen in the matrix when model.predict( ) is getting zero detections with the same 'best.pt' file

I cannot explain this ! Please help

**Comment by glenn-jocher** [comment]
Hello,

It sounds like you're encountering an intriguing issue with your model's behavior between validation and prediction. Here are a couple of things you might want to check:

1. **Ensure Consistency in Image Preprocessing**: Sometimes, discrepancies in how images are preprocessed between training/validation and inference can lead to such issues. Make sure that the same transformations and normalizations are applied.

2. **Check Confidence Thresholds**: The `model.predict()` method might be using a higher confidence threshold by default compared to what `model.val()` used. Try lowering the confidence threshold in your predict call:
   ```python
   results = model.predict('path/to/images', conf=0.25)  # Adjust the confidence threshold
   ```

3. **Model State**: Ensure that the model is correctly loading the weights and is in evaluation mode during inference:
   ```python
   model = YOLO('
